src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -335,6 +345,65 @@ void AppHost::_HandleCommandlineArgs()
     }
 }
 
+bool AppHost::_HandleLaunchArgs()
+{
+    // If someone clicks on a notification, then a fresh instance of
+    // windowsterminal.exe will spawn. We certainly don't want to create a new
+    // window for that - we only want to activate the window that created the
+    // actual notification. In the toast arg's payload will be the window id
+    // that sent the notification. We'll ask the window manager to try and
+    // activate that window ID, without even bothering to register as the
+    // monarch ourselves (if we can't find a monarch, then there are no windows
+    // running, so whoever sent it must have died.)
+    //
+    // I bet this won't work from elevated context. I'm actually quite curious.
+    // TODO!
+
+    auto activatedArgs = AppInstance::GetActivatedEventArgs();
+    if (activatedArgs != nullptr &&
+        activatedArgs.Kind() == Activation::ActivationKind::ToastNotification)
+    {
+        if (const auto& toastArgs{ activatedArgs.try_as<Activation::ToastNotificationActivatedEventArgs>() })
+        {
+            // Obtain the arguments from the notification
+            auto args = toastArgs.Argument();
+
+            // Args is gonna look like
+            //
+            // "window=id&foo=bar&..."
+            //
+            // We need to first split on &, then split those pairs on =
+
+            // tabIndex code here is left as reference for parsing multiple
+            // arguments, despite it not being used currently.
+            uint32_t window;
+            // uint32_t tabIndex = 0;
+
+            std::wstring_view argsView{ args };
+            const auto pairs = Utils::SplitString(argsView, L'&');
+            for (const auto& pair : pairs)
+            {
+                const auto pairParts = Utils::SplitString(pair, L'=');
+                if (pairParts.size() == 2)
+                {
+                    if (pairParts[0] == L"window")

```

Comment:
```suggestion
                    if (til::at(pairParts, 0) == L"window")
```
idk if it matters (because I don't understand it too well), but aren't we supposed to use til::at now?